tensorflow_probability/python/experimental/stats/sample_stats.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2018 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""Functions for computing statistics of samples."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+# Dependency imports
+from tensorflow_probability.python.stats.sample_stats import finalize_running_covariance
+from tensorflow_probability.python.stats.sample_stats import init_running_covariance
+from tensorflow_probability.python.stats.sample_stats import update_running_covariance
+
+
+__all__ = [
+    'RunningCovariance',
+]
+
+
+class RunningCovariance():
+  def __init__(self, shape, event_ndims='full'):
+    self.covariance_state = init_running_covariance(shape, event_ndims)
+
+  def update(self, new_sample, axis=None):
+    self.covariance_state = update_running_covariance(
+        self.covariance_state, new_sample, axis)
+
+  def finalize(self, ddof=0):
+    return finalize_running_covariance(self.covariance_state, ddof)
+
+  @property
+  def num_samples(self):
+    return self.num_samples
```

Comment:
surely this doesn't work?

also: this object will not be able to pass through the arguments/return values of a tf.while_loop body. I'd suggest instead just grouping the functions as a namedtuple a la:

stats/__init__.py
```
streaming_covariance = collections.namedtuple('streaming_stat', 'init,update,finalize')(
  init=init_running_covariance,
  update=update_running_covariance,
  finalize=finalize_running_covariance)
```

Then usage would be something like
```
streaming_stat = tfp.experimental.stats.streaming_covariance
_, s = tf.while_loop(
    lambda i, _: i < 10,
    lambda i, s: i+1, streaming_stat.update(s, some_val, [axis here?])
    # or make `tfp.experimental.stats.streaming_covariance(axis)` a function that returns a namedtuple with update closing over axis?
    (0, streaming_stat.init(..))
streaming_stat.finalize(s)  # again, should ddof go here or be captured by a closure?

Reply:
(by the first line, i mean surely `self.num_samples` doesn't work; the rest of the comment is higher-level; sorry if confusing)

Reply:
haha it definitely doesn't, was meant to be `self.covariance_state.num_samples`. And ohhh yeah the namedtuple approach makes sense, I'll switch it over. Thanks for the feedback Brian!